library_generation/README.md

```diff
@@ -1,235 +1,252 @@
+
+## Parameters to generate a repository using `generate_repo.py`
+
+### Generation configuration yaml (`generation_config_yaml`)
+
+A path to a configuration file containing parameters to generate the repository.
+Please refer [Configuration to generate a repository](#configuration-to-generate-a-repository) 
+for more information.
+
+### Target library API shortname (`target_library_api_shortname`), optional
 
-In order to generate a GAPIC library, you need to pull `google/` from [googleapis](https://github.com/googleapis/googleapis)
-and put it into `output` since protos in `google/` are likely referenced by 
-protos from which the library are generated.
-
-In order to generate a post-processed GAPIC library, you need to pull the
-original repository (i.e. google-cloud-java) and pass the monorepo as
-`destination_path` (e.g. `google-cloud-java/java-asset`). 
-This repository will be the source of truth for pre-existing
-pom.xml files, owlbot.py and .OwlBot.yaml files. See the option belows for
-custom postprocessed generations (e.g. custom `versions.txt` file).
-
-Post-processing makes use of python scripts. The script will automatically use
-`pyenv` to use the specified version in
-`library_generation/configuration/python-version`. Pyenv is then a requirement
-in the environment.
-
-
-## Parameters to run `generate_library.sh`
-
-You need to run the script with the following parameters.
-
-### proto_path
-A directory in `$cwd/output` and copy proto files into it. 
-The absolute path of `proto_path` is `$cwd/output/$proto_path`. 
-
-Use `-p` or `--proto_path` to specify the value.
-
-### destination_path 
-A directory within `$cwd/output`. 
-This is the path in which the generated library will reside. 
-The absolute path of `destination_path` is `$cwd/output/$destination_path`. 
-
-Use `-d` or `--destination_path` to specify the value.
-   
-Note that you do not need to create `$destination_path` beforehand.
-
-The directory structure of the generated library _withtout_ postprocessing is
-```
-$destination_path
-  |_gapic-*
-  |    |_src
-  |       |_main
-  |          |_java
-  |          |_resources
-  |       |_test
-  |_grpc-*
-  |    |_src
-  |       |_main
-  |          |_java
-  |    
-  |_proto-*
-  |    |_src
-  |       |_main
-  |          |_java
-  |          |_proto
-  |_samples
-      |_snippets
-          |_generated
+If specified, the libray whose `api_shortname` equals to `target_library_api_shortname`
+will be generated; otherwise all libraries in the configuration file will be
+generated.
+This can be useful when you just want to generate one library for debugging
+purpose.
+
+The default value is an empty string, which means all libraries will be generated.
+
+### Repository path (`repository_path`), optional
+
+The path to where the generated repository goes.
+
+The default value is the current working directory when running the script.
```

Comment:
```suggestion
The default value is the current working directory when running the script. For example, `cd google-cloud-java && python generate_repo.py ...` without specifying the `--repository_path` option will modify the `google-cloud-java` repository the user `cd`'d into.
```

Just a redundant explanation to clarify.

Reply:
done